Use ReactDOM APIs

zpao opened the pull request:
This updates ReactJS.php to require that the provided `libsrc` also has `ReactDOM` and `ReactDOMServer` exposed as globals. I think this ultimately is the best path forward for the time being.

I updated the example to build such a bundle. But I don't have PHP set up right to run this so I haven't actually tested this. Anybody want to sanity check this?

CC people who have filed issues recently @Finetuned @daslicht @tauseefk

zpao committed:
Use current React APIs

zpao committed:
Update example to build & use compiled React bundle, JSX

Finetuned commented:
@zpao Yes, that works.

```
        $ls = array();
        $ls[] = "print('RDS Keys = ' + Object.keys(global.ReactDOMServer));";
        $req = implode(";\n", $ls);
        echo $this->executeJS($req);
```

returns 

```
RDS Keys = renderToString,renderToStaticMarkup,version
```

talyssonoc commented:
Some news about when it will get merged?

18601673727 commented:
We want this got merged please.

edisonthk commented:
We need this pull request to be merged please.

stroborobo commented:
Any news on this one?

zpao commented:
Sorry for the delay. I think I might end up doing this slightly differently now that we ship react-dom-server.js which has the API we want, as opposed to building our own package (even though that is easy). I'll take a look today.

zpao committed:
Add documentation for non-bundle approach

Just in case... The bundle approach is preferred since a build step will
likely be necessary regardless and can be integrated into other
processes.
